06-wrappers/02-UiButton/components/UiButton.vue

```diff
@@ -1,10 +1,37 @@
 <template>
-  <button class="button button_secondary button_block">BUTTON</button>
+  <component :is="tag" class="button" :class="`button_${variant} ${isBlock}`" :type="type">
+    <slot></slot>
+  </component>
 </template>
 
 <script>
 export default {
   name: 'UiButton',
+  props: {
+    tag: {
+      type: String,
+      default: 'button',
+    },
+    variant: {
+      type: String,
+      default: 'secondary',
+    },
+    block: {
+      type: Boolean,
+    },
+  },
+  computed: {
+    isBlock() {
+      return this.block ? 'button_block' : '';
+    },
+    type() {
+      if (this.tag === 'button' && !this.$attrs.type) {
+        return 'button';
+      } else {
+        return this.$attrs.type;
+      }
+    },
```

Comment:
Так как от атрибута `type` здесь зависит логика работы компонента, происходят вычисления — лучше явно определить его, как параметр. И такое решение не будет работать в Composition API, так как там `$attrs` не реактивный совсем. (В Options API тоже не совсем реактивный, но сохраняет реактивность, если косвенно используется в рендеринге).

Reply:
если я вас правильно понял, то задача - по возможности уйти от использования $attrs. 
Решение: 
```suggestion
    type() {
      if (this.tag === 'button' && !this.$attrs.type) {
        return 'button';
      } else {
        return this.$attrs.type;
      }
    },
```

Reply:
Да, только с type. В оде недолжно быть работы с конкретными свойствами из атрибутов, из $attrs, только целиком передача этого объекта